### PR TITLE
Feat: add resource aws_cloud_configuration

### LINKED ISSUE #922: resource cloud_configuration
Please add resource to manage Cloud configuration for insights 
[Create API ](https://docs.env0.com/reference/cloud-resources-service-create-cloud-configuration)
[Get by Id API](https://docs.env0.com/reference/cloud-resources-service-find-cloud-configuration-by-id)
[Update API](https://docs.env0.com/reference/cloud-resources-service-update-cloud-configuration)
[Delete API](https://docs.env0.com/reference/cloud-resources-service-delete-cloud-configuration)

the schema should contain the following fields:

- `name` : string
- `provider` : enum , currently only AWS exist but in the neer future we will add more cloud providers (GCP/AZURE)
- `configuration`: according to the provider
- - AWS configuration:
- - - `accountId` -string 
- - -  `bucketName` -string
- - -  `regions` - string[], array of all the regions, if you can add a close list of all the AWS regions it will be great but don't over invest
- - - `prefix` - string (optional) 
- `health` - boolean (CALCULATED) , indicate if the configuration is valid and we get scan the cloud account

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #922

### Solution

1. Added Cloud Account APIs.
2. Added unit tests for Cloud Account APIs.
3. Added AWS cloud configuration resource.
4. Added acceptance tests for the new resource.
5. Added harness tests for the new resource.
6. Added examples for the new resource.

### CODE REVIEW

File: env0/cloud_configuration.go
Comment (TomerHeber) [inline env0/cloud_configuration.go:1]: There is alot of "golang" magic here.
But basically these functions are generic enough, that support for GCP and Azure should be almost zero effort.

File: env0/cloud_configuration.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func getCloudConfigurationFromSchema(d *schema.ResourceData, provider string) (interface{}, error) {
+	var configuration interface{}
+
+	// Find the correct type, create an instance of this type, and deserialize from the schema to the instance.
```
Comment (TomerHeber) [inline env0/cloud_configuration.go:16]: magic...

File: env0/cloud_configuration.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+func getCloudConfigurationFromApi(apiClient client.ApiClientInterface, id string) (*client.CloudAccount, error) {
+	cloudAccount, err := apiClient.CloudAccount(id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get clound configuration with id '%s': %w", id, err)
+	}
+
+	var configuration interface{}
+
+	// Find the correct type, marshal the interface to bytes, and unmarshal the bytes back to an instance of the correct type.
```
Comment (TomerHeber) [inline env0/cloud_configuration.go:40]: magic...

File: env0/cloud_configuration.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func getCloudConfigurationFromSchema(d *schema.ResourceData, provider string) (interface{}, error) {
+	var configuration interface{}
+
+	// Find the correct type, create an instance of this type, and deserialize from the schema to the instance.
+
+	switch provider {
+	case "AWS":
```
Comment (TomerHeber) [inline env0/cloud_configuration.go:19]: basically to add support for GCP or Azure, just need to add a "case" for them.

File: env0/cloud_configuration.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to get clound configuration with id '%s': %w", id, err)
+	}
+
+	var configuration interface{}
+
+	// Find the correct type, marshal the interface to bytes, and unmarshal the bytes back to an instance of the correct type.
+
+	switch cloudAccount.Provider {
+	case "AWS":
```
Comment (TomerHeber) [inline env0/cloud_configuration.go:43]: basically to add support for GCP or Azure, just need to add a "case" for them.

File: env0/resource_aws_cloud_configuration.go
Comment (TomerHeber) [inline env0/resource_aws_cloud_configuration.go:1]: Adding a GCP or Azure resource should be easy (just need to define the schema).

File: env0/errors.go [hunk 5]
```diff
@@ -11,15 +12,14 @@ import (
 )
 
 func driftDetected(err error) bool {
-	if frerr, ok := err.(*http.FailedResponseError); ok && frerr.NotFound() {
+	var failedResponseError *http.FailedResponseError
```
Comment (TomerHeber) [inline env0/errors.go:15]: This is the right way to handle errors starting go 1.13.
(Errors in golang can be wrapped within other errors, casting is no longer the correct way).

**Review by HeverFarber — APPROVED** [review]
LGMT - Maybe we can add warning (Not important here) when the health is false when we create/update  (not fail the apply) the error message under lastTest.error

**Pull request merged**